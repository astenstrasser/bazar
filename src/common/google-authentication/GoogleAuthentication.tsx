import React from "react";
import { useGoogleAuth } from "react-gapi-auth2";
import "./GoogleAuthentication.css";
import LoginIcon from "../../assets/login-icon.svg";
import LogoutIcon from "../../assets/logout-icon.svg";
import ActionIcon from "../action-icon/ActionIcon";

function GoogleAuthentication(): React.ReactElement {
  const { googleAuth } = useGoogleAuth();

  const handleSignOut = () => {
    googleAuth?.signOut();
    alert("Deslogado com sucesso!");
  };

  return (
    <div className="google-login">
      {googleAuth?.isSignedIn.get() == false && (
        <ActionIcon
          onClick={() => googleAuth?.signIn()}
          src={LoginIcon}
          alt={"Ícone para logar na aplicação"}
        />
      )}

      {googleAuth?.isSignedIn.get() && (
        <ActionIcon
          onClick={handleSignOut}
          src={LogoutIcon}
          alt={"Ícone para deslogar da aplicação"}
        />
      )}
    </div>
  );
}

export default GoogleAuthentication;
