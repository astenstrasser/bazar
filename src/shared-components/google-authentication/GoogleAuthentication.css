.action-icon {
    width: 2rem;
    cursor: pointer;
}
