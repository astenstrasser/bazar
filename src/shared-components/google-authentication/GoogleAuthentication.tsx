import React, { useContext } from "react";
import GoogleLogin, {
  GoogleLoginResponse,
  GoogleLoginResponseOffline,
  GoogleLogout,
} from "react-google-login";
import { AuthContext } from "../../App";

import "./GoogleAuthentication.css";

interface AuthProps {
  onLoginSuccess: (
    response: GoogleLoginResponse | GoogleLoginResponseOffline
  ) => void;
  onLogoutSuccess: () => void;
}

function GoogleAuthentication({
  onLoginSuccess,
  onLogoutSuccess,
}: AuthProps): React.ReactElement {
  const { isLoggedIn } = useContext(AuthContext);
  const client_id = process.env.REACT_APP_CLIENT_ID || "";

  const onFailure = () => {
    console.log("fail");
  };

  return (
    <div className="google-login">
      {!isLoggedIn && (
        <GoogleLogin
          clientId={client_id}
          buttonText="Login"
          onSuccess={onLoginSuccess}
          onFailure={onFailure}
          cookiePolicy={"single_host_origin"}
          isSignedIn={true}
        />
      )}
      {isLoggedIn && (
        <GoogleLogout
          clientId={client_id}
          buttonText="Logout"
          onFailure={onFailure}
          onLogoutSuccess={onLogoutSuccess}
        />
      )}
    </div>
  );
}

export default GoogleAuthentication;
