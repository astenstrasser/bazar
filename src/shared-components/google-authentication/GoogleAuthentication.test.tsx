import { render } from "@testing-library/react";
import React from "react";
import GoogleAuthentication from "./GoogleAuthentication";
import { GApiProvider } from "react-gapi-auth2";

test("renders snapshot", () => {
  const { container } = render(
    <GApiProvider clientConfig={{ client_id: "" }}>
      <GoogleAuthentication />
    </GApiProvider>
  );
  expect(container).toMatchSnapshot();
});
