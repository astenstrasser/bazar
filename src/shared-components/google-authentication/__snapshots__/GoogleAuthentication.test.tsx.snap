// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<div>
  <div
    class="google-login"
  />
</div>
`;
